### ISSUE TITLE
UnicodeDecodeError with keras2json

### ISSUE THREAD

**Issue by cyrilbecot** [lwtnn/lwtnn#108]
Hello,
I'm trying to run the keras2json.py script with the attached architecture to get the file I need to use my NN

While running in, I get the following issue - which I cannot reproduce when I try to json.load the attached files : 
```
Traceback (most recent call last):
  File "/lwtnn/converters/keras2json.py", line 165, in <module>
    _run()
  File "/lwtnn/converters/keras2json.py", line 46, in _run
    inputs = json.load(inputs_file)
  File "/usr/lib/python3.6/json/__init__.py", line 296, in load
    return loads(fp.read(),
  File "/usr/lib/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position
0: invalid start byte
```

Another relevant information that @dguest noticed : I did use the Sequential API, explicitely adding an InputLayer at the beginning of the network. It seems this is the origin of the problem ?

Kind regards, and thanks for the nice tool ! :)
Cheers
Cyril

[trained_model.zip](https://github.com/lwtnn/lwtnn/files/5209981/trained_model.zip)

**Comment by dguest** [comment]
Hi, can you please specify exactly the command you're running? I can get a graph out by running

```
keras2json.py  architecture-hack.json variables.json weights.h5 | sequential2graph.py
```

noting that `architecture-hack.json` has had the first `InputLayer` removed.

**Comment by dguest** [comment]
The issue that I wanted to keep track of here is that `InputLayer` isn't supported in sequential models. I'm guessing that the reason you're getting the error listed here is that `kerasfunc2json.py` and `keras2json.py` don't take the same input ordering: it's probably trying to read the `h5` file as though it's json.

**Comment by cyrilbecot** [comment]
Hey

So I use this command :
```python /lwtnn/converters/keras2json.py /out_model/architecture.json /out_model/weights.h5 /out_model/variables.json```

I just tried the command you mention and I seem to be getting the same error : 
```
Traceback (most recent call last):
  File "/lwtnn/converters/keras2json.py", line 165, in <module>
    _run()
  File "/lwtnn/converters/keras2json.py", line 46, in _run
    inputs = json.load(inputs_file)
  File "/usr/lib/python3.6/json/__init__.py", line 296, in load
    return loads(fp.read(),
  File "/usr/lib/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
Traceback (most recent call last):
  File "/lwtnn/converters/sequential2graph.py", line 70, in <module>
    run()
  File "/lwtnn/converters/sequential2graph.py", line 26, in run
    input_nn = json.load(input_file)
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

By the way this is all running in a docker image. If you want to try it out to iron out any environment inconsistency, this is what I'm running exactly :

```docker run -u $(id -u):$(id -g) --gpus all --rm -v $PWD/trained_model:/out_model -it cbecot/atlas_ssww_ml_env bash -c "python /lwtnn/converters/keras2json.py /out_model/architecture-hacked.json /out_model/weights.h5 /out_model/variables.json | /lwtnn/converters/sequential2graph.py"```

Cheers
Cyril

**Comment by cyrilbecot** [comment]
Ok, typical case of the user being the problem - arguments where not in proper order, I'll close this now and sorry for the noise...

**Comment by dguest** [comment]
Hi @cyrilbecot, just to be clear:
 * Does the model work as desired with the input layer?
 * Is the input layer _necessary_ for your model?

If the former is true I should fix #109 at some point. If the later is true I should fix #109 soon.

**Comment by cyrilbecot** [comment]
Hi @dguest 

I ran the lwtnn-test-lightweight-graph test and it seems to work well, this is the output :
```output_node:
BinaryClassificationOutputName 0.8643```

then I have another problem on the analysis side that it is consistently evaluated to 0, but I think it is an issue on the analysis framework side. I will ping you again when it has been fixed and I can compare the results I get out of keras with the InputLayer and with the lwtnn without.

The input layer is not necessary for me. It was mostly a trick to force keras to build the model before calling fit(), and plot the architecture

Cheers
